### PR TITLE
[example] Add poission disk sampling example

### PR DESCRIPTION
**Pull request by neozhaoliang**
This PR adds the code in this repo: https://github.com/taichi-dev/poisson-sampling-homework to the example list

### CODE REVIEW

**Review by jim19930609 — COMMENTED** [review]
Are you able to run it on Linux? Doesn't seem to work on my local Linux machine:

image

**Comment by neozhaoliang** [comment]
I got the same problem as you at first. Then I deleted my `ticache` folder and rebuilt taichi, it runs fine.

File: python/taichi/examples/algorithm/poisson_disk_sampling.py [hunk 1]
```diff
@@ -0,0 +1,163 @@
+"""
+Poisson disk sampling in Taichi, a fancy version.
+Based on Yuanming Hu's code: https://github.com/taichi-dev/poisson_disk_sampling
+"""
+import taichi as ti
+import taichi.math as tm
+
+ti.init(arch=ti.cpu)
```
Comment (ailzhang) [inline python/taichi/examples/algorithm/poisson_disk_sampling.py:8]: nit: is it only supported on cpu? If so, better add a comment to explain why ;)

File: python/taichi/examples/algorithm/poisson_disk_sampling.py [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+        col = tm.mix(col, cCol2, 1 - tm.smoothstep(sf, 0, -buf.x - ew))
+        col = tm.mix(col, tm.vec3(0),
+                     1 - tm.smoothstep(sf, 0, -buf.x - ew2 - ew))
+        col = tm.mix(col, cCol,
+                     1 - tm.smoothstep(sf, 0., -buf.x - ew2 - ew * 2))
+        col = tm.sqrt(ti.max(col, 0))
+        img[i, j] = col
+
+
+refresh_scene()
```
Comment (ailzhang) [inline python/taichi/examples/algorithm/poisson_disk_sampling.py:141]: nit: please add a main function for the script and doing things in the global scope in the script.

File: python/taichi/examples/algorithm/poisson_disk_sampling.py [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+gui = ti.ui.Window("Poisson Disk Sampling", res=(window_size, window_size))
+canvas = gui.get_canvas()
+gui.fps_limit = 10
+while gui.running:
+    gui.get_event(ti.ui.PRESS)
```
Comment (ailzhang) [inline python/taichi/examples/algorithm/poisson_disk_sampling.py:146]: add some comments at the beginning of the script to explain what are the supported GUI operations?

**Review by ailzhang — APPROVED** [review]
LGTM, will be nicer if you can add some comments to the code :D

File: python/taichi/examples/algorithm/poisson_disk_sampling.py [hunk 4]
```diff
@@ -0,0 +1,173 @@
...
+
+User interface:
+
+1. Click on the window to restart the animation.
+2. Press `p` to save screenshot.
+"""
+import taichi as ti
+import taichi.math as tm
+
+ti.init(arch=ti.gpu)
```
Comment (ailzhang) [inline python/taichi/examples/algorithm/poisson_disk_sampling.py:13]: @neozhaoliang these should be moved into main function as well.

**Pull request merged**